test/integration/expressions.spec.ts

```diff
@@ -625,5 +625,212 @@ describe("expressions", () => {
 
 			expect(post.id).toBeDefined();
 		});
+
+		// This test case creates an ability that uses a nested "some" clause that filters on a related model
+		it("should be able to handle expressions that are multi-level objects", async () => {
+			const initial = new PrismaClient();
+
+			const role = `USER_${uuid()}`;
+
+			const client = await setup({
+				prisma: initial,
+				customAbilities: {
+					User: {
+						customReadAbility: {
+							description: "Read user that made a post with a tag labeled with the tag context value",
+							operation: "SELECT",
+							expression: (client: PrismaClient, row, context) => {
+								return client.post.findFirst({
+									where: {
+										authorId: row("id"),
+										tags: {
+											some: {
+												label: context("ctx.label"),
+											},
+										},
+									},
+								});
+							},
+						},
+					},
+				},
+				getRoles(abilities) {
+					return {
+						[role]: [abilities.Post.create, abilities.Post.read, abilities.User.customReadAbility, abilities.Tag.read],
+					};
+				},
+				getContext: () => ({
+					role,
+					context: {
+						"ctx.label": "foo",
+					},
+				}),
+			});
+
+			const testTitle = `test_${uuid()}`;
+
+			const user1 = await adminClient.user.create({
+				data: {
+					email: `test-user-${uuid()}@example.com`,
+					posts: {
+						create: {
+							title: testTitle,
+							tags: {
+								create: {
+									label: "foo",
+								},
+							},
+						},
+					},
+				},
+			});
+
+			const user2 = await adminClient.user.create({
+				data: {
+					email: `test-user-${uuid()}@example.com`,
+					posts: {
+						create: {
+							title: testTitle,
+							tags: {
+								create: {
+									label: "bar",
+								},
+							},
+						},
+					},
+				},
+			});
+
+			const result1 = await client.user.findFirst({
+				where: {
+					id: user1.id,
+				},
+			});
+
+			expect(result1).not.toBeNull();
+
+			const result2 = await client.user.findFirst({
+				where: {
+					id: user2.id,
+				},
+			});
+
+			expect(result2).toBeNull();
+		});
+
+		// This test case creates an ability that uses multiple nested "some" clauses that span models
+		it("should be able to handle expressions that are deep multi-level objects", async () => {
+			const initial = new PrismaClient();
+
+			const role = `USER_${uuid()}`;
+
+			const testTitle = `test_${uuid()}`;
+
+			const client = await setup({
+				prisma: initial,
+				customAbilities: {
+					User: {
+						customReadAbility: {
+							description:
+								"Read user that made a post with a tag that is also attached to a post with the title context value",
```

Comment:
I'm not following this test, what is the "title context value" here?
I find it a bit difficult to understand the concept of context in this test, (could you add some comments to help with the reading @LucianBuzzo ?)

Reply:
From the comment at the top of this test case:

> This test case creates an ability that uses multiple nested "some" clauses that span models

In Yates you can provide context values for each request, via the `getContext()` option. This can then be interpolated inside the expression using the `context` method. Here we are interpolating the `ctx.title` value from the context object, using `context('ctx.title')`. 
This is a necessary part of the test, because we are testing that this interpolation functions correctly even with multiple nested `some` clauses in the query.